docs/alternatives/navigator_extension_alternative.bs

```diff
@@ -58,10 +58,30 @@ Markup Shorthands: markdown yes, css no
 
 ### {{Navigator/getEnvironmentIntegrity()}} ### {#navigator-getenvironmentintegrity}
 
-<div class="algorithm" data-algorithm="navigator-getenvironmentintegrity-alg">
+<div algorithm="navigator-getenvironmentintegrity-alg">
+    The browser has a global <dfn for="attestations sessions map">attestation sessions map</dfn>,
+    which is a [=map=] that contains each attestation session. An attestation session contains any
+    associated state with the attester for an attestation response.
+
     The <dfn method for="Navigator"><code>getEnvironmentIntegrity(|contentBinding|)</code></dfn> method, when invoked, runs these steps:
 
-    <i>Todo</i>
+    1. Let |promise| be [=a new promise=]
+    1. Run the following steps [=in parallel=]:
+          1. If |attester| is <code>undefined</code> then:
+              1. [=Reject=] |promise| and abort these steps.
+          1. Let |attesterSessions| be the [=environment=]'s [=attestations sessions map=]
+          1. Let |domain| be the top level [=origin=]
+          1. Set |session| to |attesterSessions|[|domain|]
+          1. If |session| is <code>null</code> or |session| is timed out then:
+              1. Set |session| to new session from |attester|
+              1. If |session| is <code>null</code> then:
+                  1. [=Reject=] |promise| and abort these steps
+              1. Set |attesterSessions|[|domain|] to |session|
+          1. Set |environmentIntegrityToken| to attestation from |attester| with |contentBinding|
```

Comment:
Really wasn't sure how to write this, would love some guidance.

Reply:
it's easier if you define what methods an 'attester' has exposed, with clear inputs/outputs, and you can call that from the algorithm here.

Reply:
Thanks for the link you gave for how push API does this. Made a lot of these comments a lot easier.